bndtools.m2e/src/bndtools/m2e/MavenWorkspaceRepository.java

```diff
@@ -119,13 +120,21 @@ public MavenWorkspaceRepository() {
 		};
 
 		identificationAndVersionExpressionFunction = (bsn, version) -> {
-			RequirementBuilder builder = new RequirementBuilder(BND_INFO);
-			builder.addFilter("name", bsn, version.toString(), null);
 			return combiner.or( //
 				combiner.identity(
-					CapReqBuilder.createSimpleRequirement(IdentityNamespace.IDENTITY_NAMESPACE, bsn, version.toString())
+					new RequirementBuilder(IdentityNamespace.IDENTITY_NAMESPACE) //
+						.addFilter(new FilterBuilder().and()
+							.eq(IdentityNamespace.IDENTITY_NAMESPACE, bsn)
+							.eq("version", version)
```

Comment:
```suggestion
							.eq(IdentityNamespace.CAPABILITY_VERSION_ATTRIBUTE, version)
```
This is the identity namespace.